lib/Gateway.js

```diff
@@ -25,17 +25,24 @@ var allDevices = hassDevices // will contain customDevices + hassDevices
 debug.color = 2
 
 // util function to watch for file changes, prevents the watch to be called twice
+
 const watch = (path, fn) => {
-  var lock = false
-  fs.watch(path, function () {
-    if (lock) return
+  var watcher = null
+  try {
+    watcher = fs.watch(path, e => {
+      if (e === 'rename') {
+        debug('re-setting watch on ' + path)
+        watcher.close()
+        watch(path, fn)
+      }
 
-    lock = true
-    fn()
+      fn()
+    })
+  } catch {
     setTimeout(() => {
-      lock = false
-    }, 1000)
-  })
+      watch(path, fn)
+    }, 100)
```

Comment:
I would increese the timeout at least to 1/2 seconds

Reply:
I mean one or two seconds, not half second, sorry :)

Reply:
ah, I think that's a little long. I'm sensitive to the fact that if the file is actually removed this will end up spinning pretty rapidly. 

But a few seconds means it's possible we'll actually miss some legitimate changes that happen quickly.

It'll add some complexity but perhaps I can limit re-tries instead, or implement back-off?

Reply:
The main reason I would like an higher timeout is the one I have described in the following comment. As you said this is a pretty specific use case and I don't want that this could create any problems to users that are not using this. 

We could also enable this feature with an ENV var as an option or what do you mean with 'implement back-off'?
Also retry limit could be a valid alternative.